cleanup-anaconda.sh

```diff
@@ -3,35 +3,99 @@
+for label in $(anaconda_stdout label -o litex-hub --list | grep -oP '(?<= \+ )ci-[^ ]+')
 do
-    if [[ $label != ci* ]]
+    start_section "Label $label"
+
+    #extra label check
+    if [[ "${label:0:3}" != "ci-" ]]
     then
+        echo "[CLEANUP] Omitting non-ci label $label."
+        end_section
         continue
     fi
 
-    label_info=$(anaconda -t $ANACONDA_TOKEN label -o litex-hub --show label-name 2>&1)
-    #filter first package from the label. Packages start with '+' and have too many whitespace chars (thus xargs)
-    package=$(anaconda -t $ANACONDA_TOKEN label -o litex-hub --show $label 2>&1 | grep ' + ' | cut -f2 -d+ | head -n1 | xargs)
-    #extract build timestamp
-    timestamp=$(anaconda -t $ANACONDA_TOKEN show $package 2>&1 | grep timestamp | cut -f2 -d: | xargs)
+    #get label packages
+    packages=$(anaconda_stdout label -o litex-hub --show $label | grep -oP '(?<= \+ )[^ ]+')
+
+    #check build timestamp of the first package
+    package=$(echo "$packages" | head -n1)
+    timestamp=$(anaconda_stdout show $package | grep -oP '(?<=timestamp : )[0-9]+')
+    label_date=$(date --date=@${timestamp:0:-3})
 
-    if [[ $timestamp -lt $limit_date ]]
+    if [[ $timestamp -gt $limit_date ]]
     then
-        echo "Removing old label $label with timestamp $timestamp"
-        anaconda -t $ANACONDA_TOKEN label -o litex-hub --remove $label
-    else
-        echo "Not removing label $label with timestamp $timestamp"
+        echo "[CLEANUP] Not removing label $label with timestamp $timestamp ($label_date)"
+        end_section
+        continue
     fi
 
-done
\ No newline at end of file
+    echo "[CLEANUP] Removing packages from the old label $label with timestamp $timestamp ($label_date)"
+    for package in $packages
+    do
+        metadata=$(anaconda_stdout show $package)
+
+        labels=$(echo "$metadata" | grep -oP '(?<=labels : ).+')
+        ndownloads=$(echo "$metadata" | grep -oP '(?<=ndownloads : )[0-9]+')
+        function print_result {
+            # <OS>-64/<...>.tar.bz2
+            pkg_name=$(echo $package | grep -oP '[^/]+/[^/]+.tar.bz2')
+            echo "$1;$labels;$ndownloads;$pkg_name" >> $log_file
+            echo "[CLEANUP] $1: $pkg_name"
+        }
+
+        #make sure the package doesn't have the 'main' label
+        if echo "$labels" | grep -P "'main'"
+        then
+            print_result "On main"
+            continue
+        fi
+
+        #make sure the package hasn't been downloaded too many times
+        if [[ $ndownloads -gt $download_threshold ]]
+        then
+            print_result "In use"
+            continue
+        fi
+
+        # The '--force' switch makes package removal non-interactive
+        if anaconda -t $ANACONDA_TOKEN remove --force $package
+        then
+            print_result "Removed"
+        else
+            # This might be caused by a simultaneously running cleanup script
+            print_result "Error"
```

Comment:
`Error removing`, otherwise good to merge and bump

Reply:
Used `Rm error` to not widen the result column too much.